docs/integration-testing.md

```diff
@@ -0,0 +1,70 @@
+```
+
+**Links**: [x86](https://github.com/SeleniumHQ/docker-selenium) || [ARM](https://github.com/seleniumhq-community/docker-seleniarm)
+
+Once the local Selenium instance is running, you can execute `npm run wdio -- --selenium -b 'chrome'` or `npm run wdio -- --selenium -b 'firefox'` to start running the tests. If you would like to watch the browser as the tests execute, access the Selenium VNC server [here](http://localhost:7900/?autoconnect=1&resize=scale&password=secret).
+
+To run the tests against a remote Selenium instance, such as a Selenium Grid service, use the `--selenium-host <host>` and `--selenium-port <port>` arguments.
+
+## WDIO CLI
+
+The project wraps the WDIO execution in a custom CLI. When executing WDIO, you are not running the CLI that ships with WDIO. The customer CLI arguments are defined [here](../tools/wdio/args.mjs). The custom CLI will take arguments, construct a WDIO configuration, and execute WDIO.
+
+## Writing Tests
+
+All tests should exist in the path `tests/specs` for WDIO to pick them up. Tests should be grouped by the functionality being tested or scenario being ran. WDIO will launch a new browser for each file, execute all tests of that file in the browser, and close the browser. This means more files will launch more browsers improving test isolation but also increasing the amount of time it takes to run the full test suite. Keep this in mind when writing integration tests.
+
+- **DO** create a global `describe` wrapper to hold all the tests. The name of this wrapper will be used to name the test result in SauceLabs making it easier to review test failures.
+- **DO** create a `describe` to group tests that have similar pre/post test instructions.
+- **DON'T** save state between individual test cases.
+- **DO** load a new page to start each test.
+- **DO** use `beforeEach` to instantiate variables and state before each test.
+- **DO** use `afterEach` to cleanup global state after each test.
+- **DON'T** use `beforeAll` or `afterAll`.
+- **DO** use the `it` keyword to declare a test case that has no browser limitation.
+- **DO** use the `test()` function to declare a test case that has browser limitations.
+- **DO** avoid promise chaining and use `async/await` to test asynchronous code.
+- **DO** create custom commands in `tools/wdio/plugins/jil-commands.mjs` instead of writing utility methods for common testing functionality.
+
+### Tests for specific browsers
+
+In some cases, a test may only apply to or not work expectedly in certain browser. We have created a new global `test()` that can be used in place of `it`. The `test` method takes a browser matcher and returns a higher-order function. If the browser matcher matches the browser the test is currently running inside, the higher-order function will call the `it` global method to register the test case with the WDIO framework. Below is an example.
+
+```javascript
+import Matcher from '../../tools/jil/util/browser-matcher'
```

Comment:
Does wdio `test()` work with the old `Matcher` from JIL?

Reply:
No, thanks for catching this. I forgot to update this doc.